Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs

```diff
@@ -245,12 +245,23 @@ public void TestColorTypeConverter ()
 		{
 			var converter = new ColorTypeConverter ();
 			Assert.True (converter.CanConvertFrom (typeof(string)));
-			Assert.AreEqual (Color.Blue, converter.ConvertFromInvariantString ("Color.Blue"));
-			Assert.AreEqual (Color.Blue, converter.ConvertFromInvariantString ("Blue"));
-			Assert.AreEqual (Color.Blue, converter.ConvertFromInvariantString ("#0000ff"));
-			Assert.AreEqual (Color.Default, converter.ConvertFromInvariantString ("Color.Default"));
-			Assert.AreEqual (Color.Accent, converter.ConvertFromInvariantString ("Accent"));
-			var hotpink = Color.FromHex ("#FF69B4");
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("Color.Blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("Blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("#0000ff"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("#00f"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(2.0/3.0), converter.ConvertFromInvariantString("#a00f"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 255)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 300)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 300)"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(.8), converter.ConvertFromInvariantString("rgba(0%,0%, 100%, .8)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0%,0%, 110%)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("hsl(240,100%, 50%)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("hsl(240,110%, 50%)"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(.8), converter.ConvertFromInvariantString("hsla(240,100%, 50%, .8)"));
+			Assert.AreEqual(Color.Default, converter.ConvertFromInvariantString("Color.Default"));
+			Assert.AreEqual(Color.Accent, converter.ConvertFromInvariantString("Accent"));
+			var hotpink = Color.FromHex("#FF69B4");
 			Color.Accent = hotpink;
 			Assert.AreEqual (Color.Accent, converter.ConvertFromInvariantString ("Accent"));
 			Assert.Throws<InvalidOperationException> (() => converter.ConvertFromInvariantString (""));
```

Comment:
Maybe also add some `Assert.Throws` for parentheses violations (e.g., ColorTypeConverter, L28).